### ISSUE TITLE
no-dynamic-styling false positives for arrays

### ISSUE THREAD

**Issue by idapartners** [chakra-ui/eslint-plugin-panda#155]
It seems that the fix in #143 introduced a bunch of false positives, not all of which were fixed in #151.

I haven't been able to narrow down exactly when it does and doesn't happen, but it can be in code that includes Panda but has nothing dynamic, e.g.,

```
import { StructuredText } from 'react-datocms';
import { styled } from 'styled-system/jsx';

const StructuredContent = ({content}) => {
    return (
        <StructuredText customNodeRules={[
          renderRule(isParagraph, ({ children, key }) => {
          // ^^^ complains about this entire block, inside the brackets
            return (
              <styled.div py={1} key={key}>
                {children}
              </styled.div>
            );
          })]
...
```

But it also happens in places that have nothing to do with Panda at all (though maybe have keywords that look like they do):

```
'use client';

import { Suspense } from 'react';
import { PerspectiveCamera } from '@react-three/drei';

/**
 * Common inclusions in a Three Canvas; in particular, a background color.
 * 
 * @param {import('three').Color} color
 * @returns 
 */
export function Common ({ color }) {
    return (
        <Suspense fallback={null}>
            {color && <color attach='background' args={[color]} />}
            //                                          ^^^^^ complains
            <ambientLight />
            <pointLight position={[20, 30, 10]} intensity={3} decay={0.2} />
            <pointLight position={[ -10, -10, -10]} color={'blue'} decay={'0.2'} />
            //                       ^^^. ^^^ ^^^ complains
            <PerspectiveCamera makeDefault fov={40} position={[0, 0, 450]} />
        </Suspense>
    )
};
```

**Comment by anubra266** [comment]
@idapartners Thanks for pointing it out. Should be fixed in latest version
